File: app/src/main/java/com/hqq/example/dialog/LeftDialog.java
package com.hqq.example.dialog;

import androidx.fragment.app.FragmentManager;
import android.view.Gravity;

import com.hqq.example.R;
import com.hqq.core.ui.dialog.BaseDialog;

/**
 * @version V1.0 <描述当前版本功能>
 * @author: huangqiqiang
 * 在此写用途
 * @FileName: com.hqq.core.app.dialog.RightDialog.java
 * @emain: 593979591@qq.com
 * @date: 2019-07-03 22:22
 */
public class LeftDialog extends BaseDialog {
    public static void showDialog(FragmentManager fragmentManager) {
        LeftDialog fragment = new LeftDialog();
        fragment.show(fragmentManager);
    }

    @Override
    public int getLayoutId() {
        return R.layout.dialog_right;
    }

    @Override
    public void initView() {

    }
    @Override
    public int getGravity() {
        return Gravity.LEFT;
    }

    @Override
    public int getAnimation() {
        return R.anim.dialog_left_right;
    }
}

File: app/src/main/java/com/hqq/example/dialog/LeftDialog.kt
package com.hqq.example.dialog

import android.view.Gravity
import androidx.fragment.app.FragmentManager
import com.hqq.core.ui.dialog.BaseDialog
import com.hqq.example.R

/**
 * @version V1.0 <描述当前版本功能>
 * @author: huangqiqiang
 * 在此写用途
 * @FileName: com.hqq.core.app.dialog.RightDialog.java
 * @emain: 593979591@qq.com
 * @date: 2019-07-03 22:22
</描述当前版本功能> */
class LeftDialog : BaseDialog() {
    companion object {
        fun showDialog(fragmentManager: FragmentManager?) {
            val fragment = LeftDialog()
            fragment.show(fragmentManager!!)
        }
    }

    override val layoutId: Int = R.layout.dialog_right
    override val gravity: Int = Gravity.LEFT
    override val animation: Int = R.anim.dialog_left_right

    override fun initView() {}

}